api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -1350,6 +1350,45 @@ type AWSPlatformSpec struct {
 	// +kubebuilder:default=false
 	// +optional
 	MultiArch bool `json:"multiArch"`
+
+	// SharedVPC contains fields that must be specified if the HostedCluster must use a VPC that is
+	// created in a different AWS account and is shared with the AWS account where the HostedCluster
+	// will be created.
+	//
+	// +optional
+	SharedVPC *AWSSharedVPC `json:"sharedVPC,omitempty"`
+}
+
+// AWSSharedVPC contains fields needed to create a HostedCluster using a VPC that has been
+// created and shared from a different AWS account than the AWS account where the cluster
+// is getting created.
+type AWSSharedVPC struct {
+
+	// IngressRoleARN is the ARN of a role in the VPC owner account that allows managing DNS records
+	// in the private hosted zone of the HostedCluster. The ingress operator will assume this role to
+	// manage DNS records.
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Pattern:=`^arn:(aws|aws-cn|aws-us-gov):iam::[0-9]{12}:role\/.*$`
+	// +required
+	IngressRoleARN string `json:"ingressRoleARN"`
```

Comment:
have we considered having RolesRef AWSSharedVPCRolesRef `json:"rolesRef"`?

Reply:
I've now added a struct for AWSSharedVPCRolesRef. I also documented the referenced roles in the same way we do for AWSRolesRef. Thx!